data/crac-file/crac-merging/src/main/java/com/farao_community/farao/data/crac_file/crac_merging/CracMerging.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.data.crac_file.crac_merging;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_file.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * CRAC merging function
+ *
+ * @author Viktor Terrier {@literal <viktor.terrier at rte-france.com>}
+ */
+public final class CracMerging {
+
+    private CracMerging() {
+        throw new AssertionError("Utility class should not be instanced!");
+    }
+
+    private static void addMonitoredBranch(List<MonitoredBranch> list, MonitoredBranch monitoredBranch) {
+        if (!list.contains(monitoredBranch)) {
+            list.add(monitoredBranch);
+        }
+    }
+
+    private static void addContingency(List<Contingency> list, Contingency contingency) {
+
+        List<MonitoredBranch> contingencyMonitoredBranches  = new ArrayList<>();
+
+        if (!list.contains(contingency)) {
+            list.add(contingency);
+        } else {
+            int indexCont = list.indexOf(contingency);
+            list.get(indexCont).getMonitoredBranches().forEach(mbFromList -> addMonitoredBranch(contingencyMonitoredBranches, mbFromList));
```

Comment:
It would be better to remove the usage of unmodifiable list in CRAC file API to avoid this re-creation of a contingency.

Reply:
Done, please check and resolve the conversation